test/distributed/index_map_kernels.cpp

```diff
@@ -0,0 +1,360 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/distributed/index_map_kernels.hpp"
+
+
+#include <algorithm>
+#include <memory>
+
+
+#include <gtest/gtest-typed-test.h>
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/device_matrix_data.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/matrix_data.hpp>
+#include <ginkgo/core/distributed/index_map.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+
+
+#include "core/distributed/partition_kernels.hpp"
+#include "core/test/utils.hpp"
+#include "test/utils/executor.hpp"
+
+
+using comm_index_type = gko::experimental::distributed::comm_index_type;
+
+
+class IndexMap : public CommonTestFixture {
+protected:
+    using local_index_type = gko::int32;
+    using global_index_type = gko::int64;
+    using part_type =
+        gko::experimental::distributed::Partition<local_index_type,
+                                                  global_index_type>;
+    using map_type =
+        gko::experimental::distributed::index_map<local_index_type,
+                                                  global_index_type>;
+
+    IndexMap()
+    {
+        auto connections = generate_connection_idxs(ref, this_rank, 11);
+        auto dconnections = gko::array<global_index_type>(exec, connections);
+
+        auto flat_remote_local_idxs = gko::array<local_index_type>(ref);
+        auto flat_remote_global_idxs = gko::array<global_index_type>(ref);
+        auto dflat_remote_local_idxs = gko::array<local_index_type>(exec);
+        auto dflat_remote_global_idxs = gko::array<global_index_type>(exec);
+
+        auto remote_sizes = std::vector<gko::int64>();
+
+        gko::kernels::reference::index_map::build_mapping(
+            ref, part.get(), connections, target_ids, flat_remote_local_idxs,
+            flat_remote_global_idxs, remote_sizes);
+        gko::kernels::EXEC_NAMESPACE::index_map::build_mapping(
+            exec, dpart.get(), dconnections, dtarget_ids,
+            dflat_remote_local_idxs, dflat_remote_global_idxs, remote_sizes);
```

Comment:
As you said, the first test below checks the `build_mapping` kernel. I don't use any of the output of the kernels called in the constructor, so they are completely separated. If the kernel was wrong it would not matter to the constructor, since the things defined there are only used in later tests. *Except* if the kernel segfaults. Then it would be a bit annoying

Reply:
As you said, the first test below checks the `build_mapping` kernel. I don't use any of the output of the kernels called in the constructor, so they are completely separated. If the kernel was wrong it would not matter to the constructor, since the things defined there are only used in later tests. *Except* if the kernel segfaults. Because of that I moved the test up and into a different test fixture.